lib/honeybadger/plugins/faktory.rb

```diff
@@ -0,0 +1,48 @@
+require 'honeybadger/plugin'
+require 'honeybadger/ruby'
+
+module Honeybadger
+  module Plugins
+    module Faktory
+      class Middleware
+        def call(worker, job)
+          Honeybadger.clear!
+          yield
+        end
+      end
+
+      Plugin.register do
+        requirement { defined?(::Faktory) }
+
+        execution do
+          ::Faktory.configure_worker do |faktory|
+            faktory.worker_middleware do |chain|
+              chain.prepend Middleware
+            end
+          end
+
+          ::Faktory.configure_worker do |faktory|
+            faktory.error_handlers << lambda do |ex, params|
+              opts = {parameters: params}
+
+              if job = params[:job]
+                if (threshold = config[:'faktory.attempt_threshold'].to_i) > 0
+                  retry_opt = job['retry'].to_i
+                  current_retry = job['failure']['retry_count'].to_i
```

Comment:
Could `job['failure']` ever be `nil`? This might be a good opportunity for `Hash#dig`.

Reply:
I didn't `dig` enough into faktory to be sure that it will be there on all calls. I'm assuming it will, but I'm cool with digging just to be safe.